tests/test_observability/test_scrape_metrics.py

```
@@ -240,3 +234,66 @@ def test_disabled_env_var_short_circuits(tmp_path: Path, monkeypatch) -> None:
     assert rc == 0
     assert not out.exists()
+
+
+@pytest.mark.asyncio
+async def test_records_carry_identity_labels(tmp_path: Path) -> None:
+    """``--label NAME=VALUE`` attaches a dict to every JSONL record. Downstream
+    OpenMetrics replay splices these into each metric sample so multiple runs in
+    one Prometheus TSDB stay distinguishable.
+    """
+
+    async def _handler(_req):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_handler`

(ANN202)